docs/book/src/developer/architecture/controllers/control-plane.md

```diff
@@ -233,6 +233,18 @@ The `status` object **may** define several fields:
 * `externalManagedControlPlane` - is a bool that should be set to true if the Node objects do not
   exist in the cluster. For example, managed control plane providers for AKS, EKS, GKE, etc, should
   set this to `true`. Leaving the field undefined is equivalent to setting the value to `false`.
+* `paused` - is a condition that reports if the cluster or control plane resource is paused. It should check if 'spec.paused' is set on the cluster, and for the paused annotation on the resource. This is currently optional, but will become required in future.
+```go
+	// Return early and set the paused condition to True if the object or Cluster
+	// is paused.
+	if annotations.IsPaused(cluster, m) {
```

Comment:
This snippet assumes that there is something writing the modified object in defer. While we usually do this with the patch helper:
1. it's not a contract, not sure who is currently doing this (probably most though)
2. As far as I can tell today most of our controllers are creating the patchHelper after they check for paused 

Maybe the best way to address this is to just add a note here that we assume that the change to the object has to be written, e.g. via the patchHelper in defer